### DISCUSSION TITLE
Feature: support output pchg for nspin=4

### DISCUSSION THREAD

**Discussion by WHUweiqingzhou** [deepmodeling/abacus-develop#5815] [General]
### Background

see https://mcresearch.github.io/abacus-user-guide/abacus-chg.html

![Image](https://github.com/user-attachments/assets/8deec063-d331-4b37-b6a7-fce3f46c1a37)

### Describe the solution you'd like

Please support it.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

**Comment by mohanchen** [comment]
The needed function can be realized by using PYATB package, users can try to use ABACUS+PYATB. We will transfer this issue to discussion.
